cmd/monaco/purge/purge.go

```diff
@@ -54,46 +53,39 @@ func purge(fs afero.Fs, deploymentManifestPath string, environmentNames []string
 		return errors.New("error while loading manifest")
 	}
 
-	deleteErrors := purgeConfigs(maps.Values(mani.Environments), apis)
-
-	for _, e := range deleteErrors {
-		log.WithFields(field.Error(e)).Error("Deletion error: %s", e)
-	}
-	if len(deleteErrors) > 0 {
-		return fmt.Errorf("encountered %v errors during delete", len(deleteErrors))
-	}
-	return nil
+	return purgeConfigs(maps.Values(mani.Environments), apis)
 }
 
-func purgeConfigs(environments []manifest.EnvironmentDefinition, apis api.APIs) (errors []error) {
+func purgeConfigs(environments []manifest.EnvironmentDefinition, apis api.APIs) error {
 
 	for _, env := range environments {
-		deleteErrors := purgeForEnvironment(env, apis)
-
-		if deleteErrors != nil {
-			errors = append(errors, deleteErrors...)
+		err := purgeForEnvironment(env, apis)
+		if err != nil {
+			return err
 		}
 	}
 
-	return errors
+	return nil
 }
 
-func purgeForEnvironment(env manifest.EnvironmentDefinition, apis api.APIs) []error {
+func purgeForEnvironment(env manifest.EnvironmentDefinition, apis api.APIs) error {
 	clients, err := dynatrace.CreateClientSet(env.URL.Value, env.Auth)
 
 	if err != nil {
-		return []error{
-			fmt.Errorf("failed to create a client for env `%s` due to the following error: %w", env.Name, err),
-		}
+		return fmt.Errorf("failed to create a client for env `%s` due to the following error: %w", env.Name, err)
 	}
 
 	ctx := context.WithValue(context.TODO(), log.CtxKeyEnv{}, log.CtxValEnv{Name: env.Name, Group: env.Group})
 
 	log.WithCtxFields(ctx).Info("Deleting configs for environment `%s`", env.Name)
 
-	errs := delete.AllConfigs(ctx, clients.Classic(), apis)
-	errs = append(errs, delete.AllSettingsObjects(ctx, clients.Settings())...)
-	errs = append(errs, delete.AllAutomations(ctx, clients.Automation())...)
+	deleteErrors := delete.AllConfigs(ctx, clients.Classic(), apis)
+	deleteErrors = append(deleteErrors, delete.AllSettingsObjects(ctx, clients.Settings())...)
+	deleteErrors = append(deleteErrors, delete.AllAutomations(ctx, clients.Automation())...)
 
-	return errs
+	if len(deleteErrors) > 0 {
+		log.Error("Encountered %d errors while puring configurations from environment %s, further manual cleanup may be needed. Errors:", len(deleteErrors), env.Name)
+		errutils.PrintErrors(deleteErrors)
```

Comment:
Please see the commit that introduces this change ;) 

I specifically made purge not fail/exit with non-zero status in case of delete errors because they pretty much happen every time the command runs (as we try to remove some builtin configs which always fails/only for settings we can tell if something can't be removed before we try).

A command that always fails seemed not too helpful, especially as we start using it in CI - and the idea was to have the cleanup step fail less often.

> It is always possible for the purge to attempt deletion of some default configurations
that can not be removed.
As much returning a non-zero exit code if deletions failed is not really helpful, as it will
happen in nearly all exections.